samples/connectivity/callnotification/src/main/java/com/example/platform/connectivity/callnotification/NotificationSource.kt

```diff
@@ -0,0 +1,204 @@
+) {
+
+    companion object {
+
+        const val ChannelId = "1234"
+
+        /*
+        Notification state sent via in intent to inform receiving broadcast what action the user wants to take
+         */
+        enum class NotificationState(val value: Int) {
+            ANSWER(0),
+            REJECT(1),
+            CANCEL(2),
+            CLEARED(3);
+
+            companion object {
+                fun valueOf(value: Int) = NotificationState.values().find { it.value == value }
+            }
+        }
+
+        //Get ringtone location
+        private val ringToneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_RINGTONE)
+        const val NOTIFICATION_ACTION = "NOTIFICATION_ACTION"
+
+        //Channel for on going call, this has low importance so that the notification is not always shown
+        @RequiresApi(Build.VERSION_CODES.O)
+        val notificationChannelOngoing = NotificationChannel(
+            ChannelId, "Call Notifications",
+            NotificationManager.IMPORTANCE_LOW,
+        )
+
+        //Notification channel for incoming calls. Will play ringtone, will be no dismissible and will stay on screen until user interacts with notification.
+        @RequiresApi(Build.VERSION_CODES.O)
+        fun notificationChannelIncoming(): NotificationChannel {
+            val notification = NotificationChannel(
+                ChannelId, "Call Notifications",
+                NotificationManager.IMPORTANCE_HIGH,
+            )
+            notification.importance = NotificationManager.IMPORTANCE_HIGH
+            notification.setSound(
+                ringToneUri,
+                AudioAttributes.Builder()
+                    .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+                    .setLegacyStreamType(AudioManager.STREAM_RING)
+                    .setUsage(AudioAttributes.USAGE_NOTIFICATION_RINGTONE)
+                    .build(),
+            )
+
+            return notification
+        }
+
+        @SuppressLint("NewApi")
+        val fakeCaller = Person.Builder()
+            .setName("Jane Doe")
+            .setImportant(true)
+            .build()
+    }
+
+    private val notificationManager = context.getSystemService<NotificationManager>()!!
+
+    //Intent and pending intent for full screen activity.
+    private val intent = Intent()
+        .setAction(Intent.ACTION_MAIN)
+        .setPackage(context.packageName)
+
+    private val pendingIntent =
+        PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+
+    init {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            notificationManager.createNotificationChannel(notificationChannelIncoming())
+            notificationManager.createNotificationChannel(notificationChannelOngoing)
+        }
+    }
+
+    /**
+     * Will post incoming call to sysUI
+     */
+    fun postIncomingCall() {
+        notificationManager.notify(1234, onCreateIncomingCallNotification())
```

Comment:
Since you reuse this as a notification ID, it would be less error prone if you created a constant of it and used that.